production/elasticsearch.yml

```diff
@@ -0,0 +1,105 @@
+#
+# Copyright 2017 The Jaeger Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+#
+
+# This template is based on the Elasticsearch staging example: 
+# https://github.com/kubernetes/examples/tree/master/staging/elasticsearch
+# It's not intended for production purposes, but it's simple enough for 
+# an example on our Jaeger production deployment.
+
+apiVersion: v1
+kind: Template
+labels:
+  app: jaeger
+  template: jaeger-elasticsearch-template
+  jaeger-infra: template-elasticsearch
+metadata:
+  name: jaeger-elasticsearch-template
+  annotations:
+    description: Elasticsearch storage for Jaeger
+    iconClass: icon-go-gopher
+    openshift.io/display-name: Jaeger's Elasticsearch
+    tags: instant-app,tracing,opentracing,jaeger,elasticsearch
+  labels:
+    app: jaeger
+    name: jaeger-infra
+    jaeger-infra: template-production-elasticsearch
+objects:
+- apiVersion: v1
+  kind: ServiceAccount
+  metadata:
+    name: elasticsearch
+- apiVersion: v1
+  kind: Service
+  metadata:
+    name: elasticsearch
+    labels:
+      component: elasticsearch
+  spec:
+    type: LoadBalancer
+    selector:
+      component: elasticsearch
+    ports:
+    - name: http
+      port: 9200
+      protocol: TCP
+    - name: transport
+      port: 9300
+  protocol: TCP
+- apiVersion: v1
+  kind: ReplicationController
+  metadata:
+    name: es
+    labels:
+      component: elasticsearch
+  spec:
+    replicas: 1
+    template:
+      metadata:
+        labels:
+          component: elasticsearch
+      spec:
+        serviceAccount: elasticsearch
+        containers:
+        - name: es
+          image: quay.io/pires/docker-elasticsearch-kubernetes:1.7.1-4
```

Comment:
Is this upstream elastic docker image? If not why we are not using upsream?

Reply:
What version is it using? Jaeger supports ElasticSearch 5.x.

Reply:
No idea, to be honest. I just got it from the Kubernetes repository, without much thinking: https://github.com/kubernetes/examples/tree/master/staging/elasticsearch

Reply:
please use the supported version of ES

Reply:
Agree we need it to use ES5 - if we want the ES template to be external, then eventually we might want to direct users to the kubernetes production template: https://github.com/kubernetes/examples/tree/master/staging/elasticsearch/production_cluster

Reply:
+1 about ES5, but not sure about pointing to that template, as it's also 1.7.1. Looks like our best bet will be to refer users to the helm chart for ES: https://github.com/clockworksoul/helm-elasticsearch

Reply:
Yeah, that is why I said 'eventually' :)
The reason we might want to point to the template (once updated to use ES5) is it is under Kubernetes project, so might be more official - but would be good if they also provide a helm chart.